doc/build/snippets/design.rst

```diff
@@ -0,0 +1,61 @@
+Snippets Design
+###############
+
+This page documents design considerations related to snippets.
+
+List of Design Goals
+********************
+
+As design goals, snippets are:
```

Comment:
Can/may a snippet contain piece of code that is built?

Reply:
I wonder the same thing. If it's useful, I don't see why not. The fun part about having a YAML format is that we can put whatever we want into it. Do you have a use case?

Reply:
For CDC ACM snippets we could have a code-snippet which configures and enables USB device support (new device support requires more runtime confirmation). But there are also situations where only the configuration part (Kconfig + DTS) of snippet is needed without code, for example
thingy53 + CDC ACM UART snippet for logging backend **and** USB initialization + hello_world sampe
vs.
thingy53 + CDC ACM UART snippet for logging backend + any samples/subsys/usb

Reply:
> For CDC ACM snippets we could have a code-snippet which configures and enables USB device support (new device support requires more runtime confirmation)

I think we should leave this to a future extension once the basics are in place. The snippet.yml file format is designed for future extension.